Code/Framework/AzFramework/AzFramework/Viewport/ScreenGeometry.h

```diff
@@ -68,6 +68,14 @@ namespace AzFramework
         {
         }
 
+        //! Returns the aspect ratio of the screen/viewport dimensions.
+        //! @note Ensure ScreenSize represents a valid viewport size/dimension before calling this function.
+        float AspectRatio() const
+        {
+            AZ_Assert(m_height > 0.0f, "AspectRatio called with invalid viewport size");
+            return aznumeric_cast<float>(m_width) / aznumeric_cast<float>(m_height);
+        }
```

Comment:
If width is 0 would this generate a NaN? If it does I'd recommend asserting that it is greater than zero as well.

Reply:
Agreed, not sure why before we were only checking height, I can add this

And after some thought I might move this back to a free function to make future updates to `AZ::Vector2i` easier (sorry @spham-amzn, it's just this type might not be around for much longer)